### ISSUE TITLE
Controllers auto registering available AWS certificate to the loadbalancer

### ISSUE THREAD

**Issue by adiii717** [kubernetes-sigs/aws-load-balancer-controller#2983]
**Describe the bug**
As soon as we create a new certificate with the same wildcard with a few additional more SAN in the same region, the controller start failing with this error

```
{"level":"error","ts":1674040121.375512,"logger":"controller.ingress","msg":"Reconciler error","name":"alb-ingress-internet-facing","namespace":"","error":"ingress: myapp/myapp: multiple certificates found for host: test.example.com, certARNs: [arn:aws:acm:us-west-2:XXX:certificate/05c54796-17c6-423b-a722-19ebc37b2519 arn:aws:acm:us-west-2:XXXX:certificate/c0d3007c-40b7-469b-aa8d-869dd20f885a]"}
```

**Steps to reproduce**
create and assign requires a certificate for the ingress using the following annotation
```
alb.ingress.kubernetes.io/certificate-arn: {{ quote .Values.certificate_arn }}
```
Create a new certificate in the same region with the same wildcard, but add one more different host which is not covered in the previous certificate, it can `myapp.test2.example`, now try to add new ingress or update any existing ingress it will start complaining and the certificate will automatically be associated with load balancer.

**Expected outcome**
It should only associate the certificate to the load balancer that is mentioned in the annotation.
**Or it should respect     `alb.ingress.kubernetes.io/certificate-arn: {{ .Values.certificate_arn }}`**

**Environment**

* AWS Load Balancer controller version `v2.4.5`
* Kubernetes version 24+
* Using EKS (yes/no), if so version? `v1.24.7-eks-fb459a0`

**Additional Context**:

Ingress object for LB creation
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    alb.ingress.kubernetes.io/scheme: {{ quote .Values.scheme }}
    alb.ingress.kubernetes.io/target-type: 'ip'
    alb.ingress.kubernetes.io/listen-ports: '[{"HTTP":80,"HTTPS": 443}]'
    alb.ingress.kubernetes.io/healthcheck-path: "/healthz"
    alb.ingress.kubernetes.io/group.name: alb-ingress-{{ .Values.scheme }}-{{ .Values.environment }}
    alb.ingress.kubernetes.io/ssl-redirect: "443"
    alb.ingress.kubernetes.io/certificate-arn: {{ quote .Values.certificate_arn }}
    alb.ingress.kubernetes.io/success-codes: "200,404"
    alb.ingress.kubernetes.io/load-balancer-name: {{ quote .Values.load_balancer_name }}
    alb.ingress.kubernetes.io/wafv2-acl-arn: {{ quote .Values.wafv2_acl_arn }}
    alb.ingress.kubernetes.io/load-balancer-attributes: access_logs.s3.enabled={{ .Values.log.enabled }},access_logs.s3.bucket={{ .Values.log.bucket }},access_logs.s3.prefix=k8-alb-{{ .Values.environment }}-{{ .Values.scheme }}
  labels:
    {{- include "initial-ingress.labels" . | nindent 4 }}
  name:  {{ include "initial-ingress.fullname" . }}
spec:
  ingressClassName: alb
  defaultBackend:
    service:
      name: {{ include "initial-ingress.fullname" . }}-default-svc
      port:
        number: {{ .Values.service.port }}
```

**Comment by kishorj** [comment]
@adiii717, errors indicate controller tried to auto-discover the ACM certs for your ingress. This happens if your ingress doesn't contain the certificate-arn annotation, but configuration implies HTTPS listener. There is currently a known limitation with the auto-discovery and we plan to address in v2.5.0 - issue #1598.

**Comment by adiii717** [comment]
thanks @kishorj , interesting so seems like I need to mention `certificate-arn` on every ingress, not just on the initial ingress that kicks off the load balancer creation?

**Comment by adiii717** [comment]
> thanks @kishorj , interesting so seems like I need to mention `certificate-arn` on every ingress, not just on the initial ingress that kicks off the load balancer creation?

This does not either, it does not respect even if I specify the `certificate-arn` or remove `alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS": 443}]'`, it should be fine either as I can relay on default ALB redirect to TLS.

so its the ALB controller actually 

```
{"level":"info","ts":1674039951.4423199,"logger":"controller-runtime.certwatcher","msg":"Starting certificate watcher"}
```

That seems like watch for a certificate even if I specified ARN

**Comment by kishorj** [comment]
@adiii717, the cert watcher is from the controller runtime and is for the cert used for controller webhook, not the ones from the ACM.

As for the certificate configuration, controller merges certificate-arn across all ingresses in the group. If not specified, it resorts to certificate discovery from the ACM. The issue 1598 is about supporting multiple certificates for the same host and will be available in v2.5.0. 

I'm closing the issue for now, if you have further concerns, feel free to reach out.

**Comment by adiii717** [comment]
Hmm, I think it still discovering the certificate, despite the fact that we specified the certificate ARN on every ingress, and the security team is raising concerns about the certificate.

So the load balancer controller registers the certificate that is created for prod, for example
```
*.dev.example.com => dev certificate
*.example.com => prod certificate
```
Now the issue is, the dev controller keep registering both certificates to the listener and I can see these logs

```
{"level":"info","ts":167490939,"logger":"controllers.ingress","msg":"adding certificate to listener","stackID":"alb","resourceID":"443","arn":"arn:aws:elasticloadbalancing:us-west-2xxx","certificateARN":"arn:aws:acm:us-west-2:xxxx-faa9f4a46fd6"}
{"level":"info","ts":1674909393,"logger":"controllers.ingress","msg":"added certificate to listener","stackID":"alb","resourceID":"443","arn":"arn:aws:elasticloadbalancing:us-west-2xxxe1306192cb7c553","certificateARN":"arn:aws:acm:us-west-2:xxx:certificate/xxx-7fca-4d3b-a0f1-faa9f4a46fd6"}
```

Our goal is to **"use dev specific certs"** for the dev cluster and the controller should not attach the prod certificate.

again my question is **"why controller not respecting certificate ARN"**?
